src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -127,7 +127,16 @@ protected virtual void Initialize()
             }
 
             // Bindings may use name resolution, so provide this before reading the bindings. 
-            ScriptConfig.HostConfig.NameResolver = new NameResolver();
+            var nameResolver = new NameResolver();
+
+            var dashboardString = nameResolver.Resolve("AzureWebJobsDashboard");
```

Comment:
My main point was that if you use NameResolver for this, and someone provides the value via connection strings (which is supported) you'll think the connection string is null when it isn't. So this introduces a discrepency in the way connection string are handled. It's easier to just have the same model, rather than try to remember this quirk I think.